core/internal/logger/logger.go

```diff
@@ -86,6 +87,16 @@ type ComponentLogger struct {
 	debugFlagEnabled bool
 }
 
+func init() {
+	// We do not use the internalDtLogger for this since it might never be correctly configured.
+	// We ensure this line is always printed to stderr if a development version of the library is used.
+	if version.Version == "0.0.0" {
```

Comment:
Is the FullVersion more informative here? Or will it also be 0.0.0? Because it would be interesting to have sprint + timestamp also for development versions, at least CI-built ones.

Reply:
Both Version and FullVersion will always be constant on the main branch at this moment, unless we decide to merge the changes to the source code (which changes the version constants) on the release branches back into the main branch.

If we do want to do that, it might make sense to log the full version there, otherwise not really.

Not sure what the strategy is for our other packages - I think there the story is a bit different since presumably the version comes from a resource that is packaged alongside the source code in an artifact on CI builds. It's a bit different here since we're not dealing with packaged builds but simply with source code.

Reply:
OK, no low hanging fruit here then.